tuning/trainercontroller/callback.py

```diff
@@ -59,6 +67,21 @@
 CONTROLLER_OPERATIONS_KEY = OPERATIONS_KEY
 
 
+def get_evaluator(metrics: dict) -> EvalWithCompoundTypes:
+    """Returns an evaluator that can be used to evaluate simple Python expressions."""
+    all_names = {
+        **metrics,
+        **DEFAULT_NAMES.copy(),
+    }
+    all_funcs = {
+        "abs": abs,
+        "len": len,
+        "sqrt": sqrt,
+        **DEFAULT_FUNCTIONS.copy(),
```

Comment:
I see the default functions defined here: https://github.com/danthedeckie/simpleeval/blob/2a12b5856d6f70b78dc1ac38840c80c1be6c6c4e/simpleeval.py#L315

How was the total set of all_funcs created? For example, why were abs/len/sqrt added here, while things like `pow` were excluded from the total set of all funcs?

Reply:
This is just a set of functions that we felt would be useful. Functions like `pow` can be added if necessary (with some additional checks to make sure the inputs to `pow` are "small numbers").

Reply:
That sounds reasonable to me - I agree that minimizing the list of potential supported funcs in the beginning and potentially expanding it as needed is the better approach, as it would be more difficult to remove support for funcs later on (i.e., since it could break working rules, which has some implications for the type of version bump we would need to make for the library to follow good practices)